packages/cdk/utils/src/replace-units/replace-units.ts

```diff
@@ -0,0 +1,112 @@
+import { Nullable } from '@angular-ru/cdk/typings';
+
+import { isNil } from '../is-nil';
+import { isNotNil } from '../is-not-nil';
+import { UnitsMap } from './types';
+
+const REGEXP_NUMBER: string = `[0-9]+([\.][0-9]+)?`;
+
+export function replaceUnits(text: string, unitsMap: UnitsMap): string {
+    let result: string = text;
+
+    result = replaceMultiple(result, unitsMap);
+    result = replaceSingle(result, unitsMap);
+
+    return result;
+}
+
+function replaceMultiple(value: string, unitsMap: UnitsMap): string {
+    const unitKeys: string = getUnitKeys(unitsMap);
+    const pattern: RegExp = new RegExp(
+        `(^|\\s)((${REGEXP_NUMBER})(${unitKeys}))(\\s+((${REGEXP_NUMBER})(${unitKeys})))+(\\s|$)`
+    );
```

Comment:
move to another fn and write spec for that

Reply:
for writing test this function need to be public, I want to avoid that